test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -191,6 +214,361 @@ var _ = Describe("[rfe_id:77446] LLC-aware cpu pinning", Label(string(label.Open
+				return isReady
+			}, time.Minute, time.Second).Should(BeTrue())
+			Expect(testclient.Client.List(ctx, podList, listOptions)).To(Succeed())
+			Expect(len(podList.Items)).To(Equal(1), "Expected exactly one pod in the list")
+			testpod := podList.Items[0]
+			err = getter.Container(ctx, &testpod, testpod.Spec.Containers[0].Name, cpusetCfg)
+			Expect(err).ToNot(HaveOccurred())
+			cgroupCpuset, err := cpuset.Parse(cpusetCfg.Cpus)
+			Expect(err).ToNot(HaveOccurred())
+			// fetch ccx to which cpu used by pod is part of
+			cpus, err := getCCX(cgroupCpuset.List()[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cgroupCpuset).To(Equal(cpus))
+
+			kubeletRestartCmd := []string{
+				"chroot",
+				"/rootfs",
+				"/bin/bash",
+				"-c",
+				"systemctl restart kubelet",
+			}
+			_, _ = nodes.ExecCommand(ctx, &targetNode, kubeletRestartCmd)
+			nodes.WaitForReadyOrFail("post kubelet restart", targetNode.Name, 20*time.Minute, 3*time.Second)
+			// giving kubelet more time to stabilize and initialize itself before
+			testlog.Infof("post restart: entering cooldown time: %v", restartCooldownTime)
+			time.Sleep(restartCooldownTime)
+
+			testlog.Infof("post restart: finished cooldown time: %v", restartCooldownTime)
+			// After restart of kubelet check the deployment
+			desiredStatus := appsv1.DeploymentStatus{
+				Replicas:          1,
+				AvailableReplicas: 1,
+			}
+			err = deployments.WaitForDesiredDeploymentStatus(ctx, dp, testclient.Client, testutils.NamespaceTesting, dp.Name, desiredStatus)
+			Expect(err).ToNot(HaveOccurred())
+			testlog.TaggedInfof("Deployment", "Deployment %q is running after reboot", dp.Name)
+
+			listOptions = &client.ListOptions{Namespace: testutils.NamespaceTesting, LabelSelector: labels.SelectorFromSet(dp.Spec.Selector.MatchLabels)}
+
+			Eventually(func() bool {
+				isReady, err := deployments.IsReady(ctx, testclient.Client, listOptions, podList, dp)
+				Expect(err).ToNot(HaveOccurred())
+				return isReady
+			}, time.Minute, time.Second).Should(BeTrue())
+			Expect(testclient.Client.List(ctx, podList, listOptions)).To(Succeed())
+			Expect(len(podList.Items)).To(Equal(1), "Expected exactly one pod in the list")
+			testpod = podList.Items[0]
+
+			err = getter.Container(ctx, &testpod, testpod.Spec.Containers[0].Name, cpusetCfg)
+			Expect(err).ToNot(HaveOccurred())
+			cgroupCpuset, err = cpuset.Parse(cpusetCfg.Cpus)
+			testlog.TaggedInfof("Pod", "pod %s using cpus %q", testpod.Name, cgroupCpuset.String())
+			Expect(err).ToNot(HaveOccurred())
+			// fetch ccx to which cpu used by pod is part of
+			cpus, err = getCCX(cgroupCpuset.List()[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cgroupCpuset).To(Equal(cpus))
+			testlog.TaggedInfof("L3 Cache Group", "L3 Cache group associated with Pod %s using cpu %d is: %q", testpod.Name, cgroupCpuset.List()[0], cpus)
+		})
+
+		It("[test_id:77726] Multiple Pods are not sharing same L3 cache", func(ctx context.Context) {
```

Comment:
the test name must be changed because comments at the bottom of the test acknowledge pods can indeed share the same L3 block

Reply:
Addressed in latest commit